crates/cmds-solana/src/nft/create_metadata_account.rs

```diff
@@ -1,183 +1,113 @@
+            inputs
+                .collection_details
+                .map(|size| CollectionDetails::V1 { size }),
+        );
+
+        Ok((minimum_balance_for_rent_exemption, [instruction].to_vec()))
     }
 
     #[allow(clippy::too_many_arguments)]
-    async fn command_proxy_create_metadata_accounts(
+    async fn create_metadata_accounts_with_proxy(
         &self,
         rpc_client: &RpcClient,
+        inputs: &Input,
         metadata_pubkey: Pubkey,
-        mint: Pubkey,
-        mint_authority: Pubkey,
-        _payer: Pubkey,
         proxy_authority: Pubkey,
-        authority: Pubkey,
-        name: String,
-        symbol: String,
-        uri: String,
-        creators: Vec<Creator>,
-        seller_fee_basis_points: u16,
-        _is_mutable: bool,
-        collection: Option<Pubkey>,
-        _uses: Uses,
-        _collection_details: Option<CollectionDetails>,
     ) -> crate::Result<(u64, Vec<Instruction>)> {
         let minimum_balance_for_rent_exemption = rpc_client
             .get_minimum_balance_for_rent_exemption(std::mem::size_of::<
                 mpl_token_metadata::state::Metadata,
             >())
             .await?;
 
-        let instructions = vec![create_proxy_create_metadata_instruction(
-            &proxy_authority,
-            &metadata_pubkey,
-            &mint,
-            &mint_authority,
-            &authority,
-            name,
-            symbol,
-            uri,
-            creators,
-            seller_fee_basis_points,
-            collection,
-        )];
+        let instruction = Instruction {
+            program_id: space_wrapper::ID,
+            accounts: [
+                AccountMeta::new_readonly(proxy_authority, false),
+                AccountMeta::new(metadata_pubkey, false),
+                AccountMeta::new_readonly(inputs.mint_account, false),
+                AccountMeta::new(inputs.mint_authority, true),
+                AccountMeta::new(inputs.fee_payer.pubkey(), true),
+                AccountMeta::new_readonly(mpl_token_metadata::ID, false),
+                AccountMeta::new_readonly(system_program::ID, false),
+            ]
+            .to_vec(),
+            data: ProxyCreateMetadataV3 {
+                name: inputs.metadata.name.clone(),
+                symbol: inputs.metadata.symbol.clone(),
+                uri: inputs.metadata_uri.clone(),
+                creators: inputs
+                    .creators
+                    .iter()
+                    .cloned()
+                    .map(|c| space_wrapper::state::creator::Creator {
+                        key: c.address,
+                        verified: c.verified.unwrap_or(false),
+                        share: c.share,
+                    })
+                    .collect(),
+                seller_fee_basis_points: inputs.metadata.seller_fee_basis_points,
+                collection: inputs.collection_mint_account,
+            }
+            .data(),
```

Comment:
@enzotar I think this is enough for the instruction data? Previously, there is this part after the instruction data:

https://github.com/space-operator/node-registry/blob/f6f9d6401f7a4342d4b0f2bfe05031b83b7e11a9/crates/cmds-solana/src/nft/create_metadata_account.rs#L39-L56 

Any reason why it is added?

Reply:
Ok. Not sure. Think maybe rushed to make it work